site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,168 @@
+A snapshot can have multiple references. The exact snapshot reference spec is documented at the [Spec](../spec/#snapshot-reference) page.
+Here we will provide some more explanations to the concepts in snapshot reference.
+
+### Reference Type
+
+There are clearly 2 types of snapshot reference, which are `branch` and `tag`. Their key differences are:
+
+- **New commit**: when a new snapshot is added as a child of a referenced snapshot, tag remains on the old snapshot, but branch reference moves to the child.
+
+- **Retention policy**: retention policy affects all the snapshots in a branch, but only a single tagged snapshot. (More details in the next section)
+
+### Retention Policy
+
+Iceberg offers a [snapshot expiration procedure](../spark-procedures/#expire_snapshots) to clean up snapshots that are not needed to free up storage space.
+Retention policy can be configured both globally and on snapshot reference to provide highly flexible customization to the expiration behavior.
+
+#### Global snapshot retention policy
+
+Global snapshot retention policy can be set through the following table properties:
+
+| Property                             | Default            | Description                                                   |
+| ------------------------------------ | ------------------ | ------------------------------------------------------------- |
+| history.expire.max-snapshot-age-ms   | 432000000 (5 days) | Default max age of snapshots to keep while expiring snapshots    |
+| history.expire.min-snapshots-to-keep | 1                  | Default min number of snapshots to keep while expiring snapshots |
+
+#### Snapshot reference retention policy
+
+Similarly, snapshot reference has the properties below to provide finer grain control:
+
+| Property                     | Type      | Description |
+|------------------------------|-----------|-------------|
+| **`min-snapshots-to-keep`**  | `int`     | For `branch` type only, the minimum number of snapshots to keep in a branch, default to the current value of table property `history.expire.min-snapshots-to-keep` when this value is evaluated |
+| **`max-snapshot-age-ms`**    | `long`    | The duration before a snapshot tagged or in a branch could be expired by any automatic snapshot expiration process, default to the current value of table property `history.expire.max-snapshot-age-ms` when this value is evaluated |
+
+#### Policy evaluation mechanism
+
+When a snapshot expiration process starts, it follows the steps described below:
+
+1. form an expiration candidate pool containing all snapshots
+2. for each snapshot reference, evaluate the associated policy and move snapshots out of the candidate pool
+3. apply global retention policy to and move snapshots out of the candidate pool
+4. when multiple snapshots can be chosen to be moved out, newer snapshots win
+4. after evaluation, expire all snapshots that are still in the candidate pool
```

Comment:
I find this description of the process confusing. Why build a pool and remove, rather than building a set of snapshots to retain? I also don't think it makes sense to refer to applying global policy as a separate step or "choosing" snapshots from the pool. Instead, I think it is more clear to use a procedure that focuses on the snapshots that must be kept:

> 1. Start with an empty set of snapshots to retain
> 2. Remove any refs (other than `main`) where the referenced snapshot is older than `max-ref-age-ms`
> 3. For each branch and tag, add the referenced snapshot to the retained set
> 4. For each branch, add its ancestors to the retained set until:
>     * The snapshot is older than `max-snapshot-age-ms`, AND
>     * The snapshot is not one of the first `min-snapshots-to-keep` in the branch (including the branch's referenced snapshot)
> 5. Expire any snapshot not in the set of snapshots to retain.
> 
> **Note:** `max-ref-age-ms`, `max-snapshot-age-ms`, and `min-snapshots-to-keep` can be set in ref metadata. If not set, the ref should use default values from table metadata.